src/libraries/Common/src/Interop/Unix/System.Native/Interop.GNameUName.cs

```diff
@@ -0,0 +1,94 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.InteropServices;
+using System.Buffers;
+using System.Text;
+using System;
+using System.Collections.Generic;
+
+internal static partial class Interop
+{
+    internal static partial class Sys
+    {
+        /// <summary>
+        /// Gets the group name associated to the specified group ID.
+        /// </summary>
+        /// <param name="gid">The group ID.</param>
+        /// <returns>On success, return a string with the group name. On failure, returns a null string.</returns>
+        internal static string GetGName(uint gid) => GetGNameOrUName(gid, isGName: true);
+
+        /// <summary>
+        /// Gets the user name associated to the specified user ID.
+        /// </summary>
+        /// <param name="uid">The user ID.</param>
+        /// <returns>On success, return a string with the user name. On failure, returns a null string.</returns>
+        internal static string GetUName(uint uid) => GetGNameOrUName(uid, isGName: false);
+
+        private static string GetGNameOrUName(uint id, bool isGName)
+        {
+            int bufferSize = 256; // Upper limit allowed for login name in kernel
```

Comment:
> // Upper limit allowed for login name in kernel

The kernel doesn't store names. This is the limit used by glibc on Linux for names.
Note that this is the limit for the name. We're actually passing this buffer to the getpwuid_r, getpwnam_r which stores more than the name in it.